docs/content/guides/dagster/ml-ops.mdx

```diff
@@ -0,0 +1,221 @@
+- Dagster models data assets, not just tasks, so it understands the upstream and downstream data dependencies and is a one-stop shop for both the data transformations and the models that depend on the data transformations.
+
+---
+
+## Machine learning development
+
+Before we go into how to operatialize a machine learning model, let's go over some of the tools you can use in Dagster to develop a machine learning model.
+
+If you are already using Dagster for your ETL pipelines, it is a natural progression to build out and test your models in Dagster.
+
+For this guide, we will be using the hackernews data which is demoed in the [tutorial](/tutorial).
+
+The machine learning model we will walk through is taking the hackernews stories, using the title to predict the number of comments that a story will generate. This will be a supervised model since we have the number of comments for all the previous stories.
+
+The assets graph will look like this at the end of this guide:
+
+<Image
+alt="alt"
+src="/images/guides/ml-ops/ml_asset_dag.png"
+width={1804}
+height={1064}
+/>
+
+### Data Ingestion
+
+First, we will be creating an asset which retrieves all the hackernews records from the previous run to the latest hackernews record.
+
+```python file=/guides/dagster/ml_ops/ml_ops.py startafter=data_ingestion_start endbefore=data_ingestion_end
+import requests
+from dagster import asset, FreshnessPolicy
+import pandas as pd
+
+@asset(freshness_policy=FreshnessPolicy(maximum_lag_minutes=10))
+def hackernews_stories():
+    """Get the max ID number from hacker news"""
+    latest_item = requests.get(
+        f"https://hacker-news.firebaseio.com/v0/maxitem.json"
+    ).json()
+    """Get items based on story ids from the HackerNews items endpoint"""
+    results = []
+    scope = range(latest_item-1000, latest_item)
+    for item_id in scope:
+        item = requests.get(
+            f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json"
+        ).json()
+        results.append(item)
+
+    df = pd.DataFrame(results)
+    if len(df) > 0:
+        df = df[df.type == "story"]
+        df = df[~df.title.isna()]
+
+    return df
+```
+
+### Data Transformation
+
+Now that we have a dataframe with all valid stories, we want to transform that data into something our machine learning model will be able to use.
+
+The first step is taking the dataframe and splitting into a [training and test set](https://en.wikipedia.org/wiki/Training,\_validation,\_and_test_data_sets). In some of your models, you also might choose to have an additional split for a validation set. The reason we split the data is so that we can have a test and/or a validation dataset that is independent of the training set and we can use that to see how well our model did.
```

Comment:
```suggestion
The first step is taking the dataframe and splitting into a [training and test set](https://en.wikipedia.org/wiki/Training,\_validation,\_and_test_data_sets). In some of your models, you also might choose to have an additional split for a validation set. The reason we split the data is so that we can have a test and/or a validation dataset that is independent of the training set. We can then use that dataset to see how well our model did.
```

🔧 Suggesting you break up the last sentence in this paragraph. It currently reads a bit like a run on and can be difficult to follow.